packages/cli-tool/src/template/fetchingStrategy/downloadStrategy.ts

```diff
@@ -0,0 +1,76 @@
+import { CliUx } from '@oclif/core';
+
+import { FetchStrategy } from '.';
+import { InitTemplateOptions } from '../.';
+import runCommand from '../../helpers/child-process';
+import { downloadRepository } from '../../helpers/github';
+
+const TEMPLATE_OWNER = 'nimblehq';
+const TEMPLATE_REPO = 'react-templates';
+
+class DownloadStrategy implements FetchStrategy {
+  constructor(public selectedTemplate: string) {
+    this.selectedTemplate = selectedTemplate;
+  }
+
+  async fetchTemplateFiles(options: InitTemplateOptions): Promise<void> {
+    return this.downloadTemplateRepository(options)
+      .then(() => this.extractDownloadedTemplateFolder(options))
+      .then(() => this.renameFolder(options))
+      .then(() => this.cleanTemporaryFiles(options));
```

Comment:
```suggestion
    await this.downloadTemplateRepository(options);
    await this.extractDownloadedTemplateFolder(options);
    await this.renameFolder(options);
    return await this.cleanTemporaryFiles(options);
```

Long time I haven't dealt with async JS, but as far as I remember, it acts the same but avoids the chained call (stack trace remains cleaner).

Reply:
Sure, I wanted to keep changes to a minimum to make it easier to review, however I can change this 👍 😄

Reply:
Fixed in 9f8df06823d65a3a35a3c0206a745da96169353b @malparty 🙌